From e1125030e89a8f385c54e93d9dffc000b8e1fdb9 Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Tue, 8 Mar 2022 23:50:18 +0100
Subject: [PATCH] Disable Recovery Patch via BOARD_NO_RECOVERY_PATCH

Add BOARD_NO_RECOVERY_PATCH := true in your BoardConfig.mk to
disable the creation of a recovery patch in
/system|vendor/etc/recovery.img.

Change-Id: Ibf7c2d695b35aad22cd97e8c67e3aa3b9629b986
Signed-off-by: Dominggoes Isakh <drjisakh@gmail.com>
---
 core/Makefile                             | 3 +++
 tools/releasetools/make_recovery_patch.py | 5 +++++
 tools/releasetools/non_ab_ota.py          | 4 ++--
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 80abd1936b..58689ffa15 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -5378,6 +5378,9 @@ endif
 ifdef BOARD_USES_VENDORIMAGE
 	$(hide) echo "board_uses_vendorimage=true" >> $@
 endif
+ifdef BOARD_NO_RECOVERY_PATCH
+	$(hide) echo "board_no_recovery_patch=true" >> $@
+endif
 ifeq ($(BOARD_AVB_ENABLE),true)
 ifeq ($(BUILDING_VBMETA_IMAGE),true)
 	$(hide) echo "avb_building_vbmeta_image=true" >> $@
diff --git a/tools/releasetools/make_recovery_patch.py b/tools/releasetools/make_recovery_patch.py
index 1497d69ed7..1396798cb2 100644
--- a/tools/releasetools/make_recovery_patch.py
+++ b/tools/releasetools/make_recovery_patch.py
@@ -49,6 +49,11 @@ def main(argv):
 
   board_uses_vendorimage = OPTIONS.info_dict.get(
       "board_uses_vendorimage") == "true"
+  if not board_no_recovery_patch:
+    if board_builds_vendorimage:
+      target_files_dir = "VENDOR"
+    elif not board_uses_vendorimage:
+      target_files_dir = "SYSTEM"
 
   if board_uses_vendorimage:
     target_files_dir = "VENDOR"
diff --git a/tools/releasetools/non_ab_ota.py b/tools/releasetools/non_ab_ota.py
index c4fd809447..b5ce9c600c 100644
--- a/tools/releasetools/non_ab_ota.py
+++ b/tools/releasetools/non_ab_ota.py
@@ -664,9 +664,9 @@ def _WriteRecoveryImageToBoot(script, output_zip):
 def HasRecoveryPatch(target_files_zip, info_dict):
   board_uses_vendorimage = info_dict.get("board_uses_vendorimage") == "true"
 
-  if board_uses_vendorimage:
+  if not board_no_recovery_patch:
     target_files_dir = "VENDOR"
-  else:
+  elif not board_uses_vendorimage:
     target_files_dir = "SYSTEM/vendor"
 
   patch = "%s/recovery-from-boot.p" % target_files_dir
-- 
2.39.2

